src/main/java/com/example/wdmsystem/customer/system/CustomerService.java

```diff
@@ -20,24 +23,91 @@ public CustomerService(ICustomerRepository customerRepository, IReservationRepos
     }
 
     public Customer createCustomer(CustomerDTO createCustomer) {
-        return null;
+        Customer customer = new Customer(
+                1,
+                createCustomer.firstName(),
+                createCustomer.lastName(),
+                createCustomer.phone(),
+                LocalDateTime.now(),
+                LocalDateTime.now()
+        );
+
+        return customerRepository.save(customer);
     }
 
-    public List<Customer> getCustomers(Date createdAtMin, Date createdAtMax, int limit) {
-        return null;
+    public List<Customer> getCustomers(LocalDateTime createdAtMin, LocalDateTime createdAtMax, Integer limit) {
+        if (createdAtMin != null && createdAtMax != null && createdAtMin.isAfter(createdAtMax)) {
+            throw new InvalidInputException("createdAtMin must be earlier than createdAtMax.");
+        }
+
+        LocalDateTime lowerDate = LocalDateTime.of(0,1,1,0,1);
+        LocalDateTime upperDate = LocalDateTime.of(9999,12,30,23,59);
+        if (createdAtMin != null) {
+            lowerDate = createdAtMin;
+        }
+        if (createdAtMax != null) {
+            upperDate = createdAtMax;
+        }
+
+        if (limit == null || limit <= 0 || limit > 250) {
+            limit = 50;
+        }
+        PageRequest pageRequest = PageRequest.of(0, limit);
+        return customerRepository.findCustomersWithinDateRange(lowerDate, upperDate, pageRequest);
     }
 
     public Customer getCustomer(int id) {
-        return null;
+        return customerRepository.findById(id)
+                .orElseThrow(() -> new NotFoundException("Customer with id " + id + " not found"));
     }
 
+    @Transactional
```

Comment:
I don't think we need this tag as Transactional management is not enabled on our config.